pkg/controllers/monitoring/monitoring-stack/components.go

```diff
@@ -23,62 +23,63 @@ const PrometheusUserFSGroupID = 65534
-		reconciler.NewUpdater(newPrometheusRole(ms, prometheusRBACResourceName, rbacVerbs), ms),
-		reconciler.NewUpdater(newRoleBinding(ms, prometheusRBACResourceName), ms),
+		reconciler.NewUpdater(newServiceAccount(prometheusName, ms.Namespace), ms),
+		reconciler.NewUpdater(newPrometheusRole(ms, prometheusName, rbacVerbs), ms),
+
+		// create clusterrole if there is a nsSelector otherwise delete any that was created earlier
+		reconciler.NewOptionalUpdater(newClusterRoleBinding(ms, prometheusName), ms, hasNsSelector),
+		reconciler.NewOptionalUpdater(newRoleBindingForRoleType(ms, prometheusName, "ClusterRole"), ms, !hasNsSelector),
+
 		reconciler.NewUpdater(newAdditionalScrapeConfigsSecret(ms, additionalScrapeConfigsSecretName), ms),
-		reconciler.NewUpdater(newServiceAccount(alertmanagerRBACResourceName, ms.Namespace), ms),
-		reconciler.NewOptionalUpdater(newAlertManagerRole(ms, alertmanagerRBACResourceName, rbacVerbs), ms,
+		reconciler.NewUpdater(newServiceAccount(alertmanagerName, ms.Namespace), ms),
+		reconciler.NewOptionalUpdater(newAlertManagerRole(ms, alertmanagerName, rbacVerbs), ms,
 			!ms.Spec.AlertmanagerConfig.Disabled),
-		reconciler.NewOptionalUpdater(newRoleBinding(ms, alertmanagerRBACResourceName), ms,
+		reconciler.NewOptionalUpdater(newRoleBindingForRoleType(ms, alertmanagerName, "Role"), ms,
 			!ms.Spec.AlertmanagerConfig.Disabled),
-		reconciler.NewOptionalUpdater(newAlertmanager(ms, alertmanagerRBACResourceName, instanceSelectorKey, instanceSelectorValue), ms,
+		reconciler.NewOptionalUpdater(newAlertmanager(ms, alertmanagerName, instanceSelectorKey, instanceSelectorValue), ms,
 			!ms.Spec.AlertmanagerConfig.Disabled),
 		reconciler.NewOptionalUpdater(newAlertmanagerService(ms, instanceSelectorKey, instanceSelectorValue), ms,
 			!ms.Spec.AlertmanagerConfig.Disabled),
 		reconciler.NewOptionalUpdater(newAlertmanagerPDB(ms, instanceSelectorKey, instanceSelectorValue), ms,
 			!ms.Spec.AlertmanagerConfig.Disabled),
-		reconciler.NewUpdater(newPrometheus(ms, prometheusRBACResourceName, additionalScrapeConfigsSecretName, instanceSelectorKey, instanceSelectorValue), ms),
+		reconciler.NewUpdater(newPrometheus(ms, prometheusName, additionalScrapeConfigsSecretName, instanceSelectorKey, instanceSelectorValue), ms),
 		reconciler.NewUpdater(newPrometheusService(ms, instanceSelectorKey, instanceSelectorValue), ms),
 		reconciler.NewUpdater(newThanosSidecarService(ms, instanceSelectorKey, instanceSelectorValue), ms),
 		reconciler.NewOptionalUpdater(newPrometheusPDB(ms, instanceSelectorKey, instanceSelectorValue), ms,
 			*ms.Spec.PrometheusConfig.Replicas > 1),
 	}
 }
 
-func newPrometheusRole(ms *stack.MonitoringStack, rbacResourceName string, rbacVerbs []string) *rbacv1.Role {
-	return &rbacv1.Role{
+func newPrometheusRole(ms *stack.MonitoringStack, rbacResourceName string, rbacVerbs []string) *rbacv1.ClusterRole {
+	return &rbacv1.ClusterRole{
```

Comment:
@JoaoBraveCoding here is the change. 
This reminds me to write better commit message which I will do in `Squash & Merge` message